src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,951 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
```

Comment:
This PR needs documentation before merging.

Reply:
We also need to add Olivia's examples to the docs.

Reply:
<details>
<summary>Olivia's Presets Notes</summary>

```
Presets
1 -Natural numbers – Primes in a spiral with Numbers
-Text: These are the natural number put in a spiral, with each of the prime numbers

highlighted.
2 -Natural numbers – Primes in a spiral
-Text: When there are lots of numbers, it becomes evident that the primes form long
diagonal lines. These diagonal lines are quadratic equations, namely x^2 + c, x^2 + 2x + c, x^2 -2x + c,
and x^2 + 4x + c.
3 -Natural numbers – Primes in rows with numbers
-Text: These are the natural numbers put in rows, with each of the prime numbers

highlighted.
4 -Primes in rows
5 -Natural numbers – Primes in rows that reset and augment with numbers
-Text: These are the natural numbers put in rows, but each row the sequence resets with

each number one more than the number above it.
6 -Natural numbers – Primes in rows that reset and augment
7 -Natural numbers – N^2 Primes in rows that reset and augment with numbers
-Text: These are the square numbers put in rows, but each row the sequence resets with

each number one more than the number above it.
8 -Natural numbers – N^2 Primes in rows that reset and augment
-Text: When there are lots of numbers, long diagonal lines become apparent. The diagonal
lines that go up and to the right each have a corresponding diagonal line that goes down and to the right
because numbers repeat when they are arranged this way. The quadratic equation for which each of these
diagonals corresponds is x^2 – x + C
9 -Natural numbers – Abundant numbers in spiral
-Text: These are the natural numbers in a spiral. The abundant numbers are the black
squares, and the non-abundant numbers are the white squares. Abundant numbers are numbers for which
the sum of the factors is greater than the As you can see, the abundant numbers have a lot more structure
than the abundant numbers.
10 -Natural numbers – Abundant numbers in rows
-Text: The structure of the abundant numbers becomes even more evident when they are

put in rows.
11 -Natural numbers – Abundant numbers and primes in spiral
-Text: The abundant numbers are black, and the primes are red. When they are show

together, primes apear less sporadic as they seems to fit around the abundant numbers.
12 -Natural numbers – Polygonal Numbers in spiral
-This is the natural numbers in a spiral. The polygonal number are highlighted. The triangle
numbers are red, the square number are orange, the pentagonal numbesr are yellow, the hexagonal
numbers are green, the heptagonal numbers are blue, and the octagonal numbers are purple. Polygonal
numbers are the number of dots that can be arranged in the shape of that polygon. For example, 6 is a
triangle number because one can form a triangle with three dots at the bottom, two dots in the middle, and
one dot at the top.
13 -Natural numbers – Polyongal Number in rows
-These are the polygonal numbers highlighted in rows.
14 -Natural numbers – N^2 Polygonal Numbers in rows that reset and augment
-These are the polygonal numbers highlighted in rows with the top row being x^2 and each

of the squares being one more than the number above it.
15 -Natural numbers, Rainbow Colored in rows with numbers
-These are the natural numbers in rows wiht each digit colored a different color.
16 -Natural numbers, Rainbow Colored in spiral
-These are the natural numbers in a spiral with each digit colored a different color.
17 -Pi (A000796) – Rainbow Colored in rows (With 1000 numbers)
18 -Pi (A000796) – Rainbow Colored in rows (With 40000 numbers)
-These are the digits of pi in rows, with each digit colored a different color.
19 -Abundant Numbers (A005101) Colored By Last Digit Alternative in a spiral

-When the abundant numbers are put in a spiral and highlighted by their last digit, the

scarcit of odd abundant numbers becomes visually apparent. (With 1000 numbers)
20 -Abundant Numbers (A005101) Colored By Last Digit Alternative in a spiral (With 40000
numbers)

-When the abundant numbers are put in a spiral and highlighted by their last digit, their

structure becomes clealy not random.

Idea for preset that can't be implemented yet:
-Natural numbers – Composite (Dark Green), Abundant (Green), Perfect (Lime Green) Primitive
Abundant (Pink), Weird (Magenta), Highly Abundant (Small Dark Blue), Superabundant (Small Blue),
Colosally Abundant (Small Cyan)

-These are various properties related to divisibilility. A number is abundant if the sum of its
factors is greater than the number itself. The way that these numbers are highlighted emphasizes the
overlap of these properties. All perfect numbers are abundant, and all abundant numbers are composite.
Similarly, all colossally abundant numbers are superabundant, and all suberabundant numbers are highly
abundant. Lastly, all weird numbers are primitive abundant and all primitive abundant numbers are
abundant.
```

</details>

Reply:
commit [89598f8](https://github.com/numberscope/frontscope/pull/230/commits/89598f88f60e344fb611a51c28af0c9861c69a0d) puts in the documentation, with the exception of the images of the examples (so this isn't resolved yet, but closer)

Reply:
OK, took a look. Some thoughts/suggestions; some of these are a matter of taste, so feel free to take or leave. 

1. I think the parameters should go before the examples, esp since the example list will be very long once the pictures are there.
2. I think the credit should go after the examples, in a general spirit of modesty.
3. Formulas in examples 8 and 14 not yet set with MathJax
4. The headers break weirdly in examples 19 and 20
5. Do we really need 20 examples? Should we do a bit of judicious editorial pruning? (If they are all really cool, of course they all can stay.)
6. The link to the properties section is broken, and there appears to be a stray attempt at putting an ID on that heading.
7. The names of the properties in camelCase are not how they appear in the dialog. They should match.
8. I _think_ that the Rows_Offset re-starts the sequence from the specified starting index, but adds one more each row to the _values_ of the sequence. That should be checked and if I am right, the documentation should be updated.
9. I suggest we use "you" rather than "the user" in these docs, it's more friendly, I think.
10. Maybe want to mention that the "numberColor" parameter is only shown when show numbers is checked.
11. The header for "backgroundColor" breaks weirdly.
12. I think there is no longer a "None" property, since it was useless anyway.
13. Documentation for the "Property N Visualization" and "Property N Color" parameters is missing.

Reply:
Thanks!  Some quick clarifications/questions: 
(2) and in general after the parameters (for other visualizers that have no examples section) -- so in the source file at the very end?
(5) good point; I was just copying the original list.  I'll wait until the images are in and see if some can be combined/eliminated
(6) odd, I thought I fixed that!
(7) I was following the format of the ShowFactors visualizer when I did this, which matched the internal variables.  (Or maybe ShowFactors actually has the same internal variables as dialog names?)
Thanks again, very helpful.

Reply:
Also, @gwhitney there's a question in Slack about where to store the image files for docs in the directory tree.  Should they go in with the source (where the source for the relevant doc is) or outside ```src``` over in ```assets``` or something?

Reply:
On (2), that's my feeling.
On (6), it might again stem from the fact that only the "staged" changes to your files get committed; the current versions of files in your working directory might differ from the "staged" versions.
On (7) dunno what ShowFactors does, I may have erred. But clearly documentation should match the identifiers visible to the person reading the docs.

On location of image files, seems like there's a precedent for src/assets/(VISUALIZER_NAME); I don't see any particular reason to change that pattern now, but if there's a reason to prefer a different location, I am not wedded to it, either.

Reply:
[newest commit](https://github.com/numberscope/frontscope/pull/230/commits/c2de3a2778875ce18ec04e6ce65438a30ae33691) should handle all of this, I hope!

Reply:
Will look again when the images/warnings are fixed.